src/clr-addons/readonly/readonly.directive.ts

```diff
@@ -0,0 +1,112 @@
+import { AfterViewChecked, Directive, ElementRef, Injector, Input, Renderer2 } from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+@Directive({
+  selector: '[clrReadonly]',
+})
+export class ClrReadonlyDirective implements AfterViewChecked {
+  @Input('clrMulti') isMultiSelect: boolean = false;
+  @Input('clrUnitPosition') unitPosition: 'left' | 'right' = 'right';
+  @Input('clrReadOnlyProperty') arrayPosition: string | null = null;
+  @Input('clrReadonly') clrReadOnly: boolean = true;
+
+  private readonly originalElement: HTMLElement;
+  private isInitialized = false;
+  private isList: boolean = false;
+  private isNumeric: boolean = false;
+  private ngControl: NgControl;
+
+  constructor(
+    private elementRef: ElementRef,
+    private readonly renderer: Renderer2,
+    private readonly injector: Injector
+  ) {
+    this.originalElement = elementRef.nativeElement;
+  }
+
+  ngAfterViewChecked(): void {
+    this.ngControl = this.injector.get(NgControl, null);
+    if (!this.isInitialized && this.ngControl.value != null) {
+      this.isInitialized = true;
+      this.renderAsSpan();
+    }
+  }
+
+  private renderAsSpan(): void {
+    const parentElement = this.originalElement.parentElement;
+    if (!parentElement) {
+      return;
+    }
+
+    this.handleDifferentControlTypes();
+
+    // Create a new span element to display the readonly value.
+    const span = this.renderer.createElement('span');
+
+    const formattedValue = this.formatControlValue(this.ngControl);
+    const textNode = this.renderer.createText(formattedValue);
+
+    // Add text and classes to the span element.
+    this.renderer.appendChild(span, textNode);
+    this.renderer.setAttribute(span, 'class', 'clr-readonly');
+    this.renderer.addClass(parentElement, 'clr-readonly-parent');
+
+    // Hide all child elements of the parent and append the new span.
+    this.hideAllChildren(parentElement);
+    this.renderer.setStyle(this.originalElement, 'display', 'none');
+    this.renderer.appendChild(parentElement, span);
+  }
+
+  private hideAllChildren(parentElement: HTMLElement): void {
+    Array.from(parentElement.childNodes).forEach(childNode => {
```

Comment:
childNodes already provides a forEach method, why converting it to an array first?